### ISSUE TITLE
Create YOLO object from torch.nn.Model

### ISSUE THREAD

**Issue by EchoLynx** [ultralytics/ultralytics#17277]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'd like to be able to feed my trained yolov8s model into a third party quantization process that consumes a torch.nn.Module. First step is getting torch.nn.Model of from my yolo object, which I can do using .model, like so:

```python
from ultralytics import YOLO
model = YOLO("best.pt")
model.model ## <<<<< This is a torch.nn.Module that I can feed into the quantization process
```

Tricky part is, I need to be able to go back from a torch.nn.Module to a YOLO object.

The quantization process is iterative, and it requires an custom evaluation function that is used to check to see what the performance of the model after each iteration. I want to be able to put the torch.nn.Module back into a YOLO object after every iteration so that I can use the YOLO object's validation feature to generate statistics to inform the next quantization iteration.

The following is pseudocode, but hopefully illustrates what I mean by this.

```python
from ultralytics import YOLO
from quantizer import *

# Load my trained model into a YOLO object
initial_yolo_model = YOLO("best.pt")
# Extract the torch.nn.Module from the YOLO object
initial_yolo_model_as_torch_module = initial_yolo_model.model
# Feed the torch.nn.Module into the quantizer along with the evaluation function
quantized_model = quantizer(initial_yolo_model_as_torch_module, evaluation_function)

# Define the evaluation function
def evaluation_function(current_model: torch.nn.Module) -> float:
    temp = YOLO(current_model) # <<< Seems like I can't do this with YOLO currently.
    return temp.val().fitness # Could use things other than fitness, but the idea here is that I want to use the val function
```

As far as I can tell, there's no YOLO constructor that takes in a torch.nn.Module.

Is there some other way I can wrap the torch.nn.Module in a YOLO object? Maybe by updating an existing object with a new torch.nn.Module? Does that desynchronize the module from the metadata in the YOLO object?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @EchoLynx, and thank you for reaching out to Ultralytics 🚀! We appreciate your interest and efforts in using our tools for your project. An Ultralytics engineer will join the conversation soon to assist you further.

In the meantime, your exploration into feeding a `torch.nn.Module` into a YOLO object sounds very interesting! To enable our team to assist you efficiently, if you are experiencing any unexpected behavior or potential bugs, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will help us to understand your issue fully.

For learning and experimentation, feel free to dive into our [Docs](https://docs.ultralytics.com) for comprehensive guides and examples of using YOLO with [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/).

In addition to direct assistance, the Ultralytics community is a great place to exchange ideas and get tips. Join our [Discord](https://ultralytics.com/discord) 🎧 for chat, visit our [Discourse](https://community.ultralytics.com) for detailed discussions, or share insights on our [Subreddit](https://reddit.com/r/ultralytics) with fellow users.

## Upgrade

Ensure you're using the latest version of the `ultralytics` package, which may contain fixes or features relevant to your work. You can upgrade with:

```bash
pip install -U ultralytics
```

This will include all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) and support for [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

Consider running YOLO in any of the following verified environments for optimal performance:

- **Notebooks** with GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Follow the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS**: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Explore our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

You can verify the current status of YOLO operations via our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule). The badge status below confirms the recent integrity of all tasks across different platforms.

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

We look forward to helping you further with your inquiry.

**Comment by EchoLynx** [comment]
> You can try assigning `quantized_model.model` to `initial_model.model.model` .

Thank you. This is the answer I was looking for.

I ended up going a different direction, but I appreciate the response nonetheless.
